qt/widgets/common/src/WorkspacePresenter/WorkspaceTreeWidget.cpp

```diff
@@ -206,28 +197,29 @@ StringList QWorkspaceDockView::getSelectedWorkspaceNames() const {
 /** Returns a pointer to the selected workspace (the first if multiple
 *   workspaces selected)
 */
-Mantid::API::Workspace_sptr QWorkspaceDockView::getSelectedWorkspace() const {
+Mantid::API::Workspace_sptr WorkspaceTreeWidget::getSelectedWorkspace() const {
   auto items = m_tree->selectedItems();
   auto data = items[0]->data(0, Qt::UserRole).value<Workspace_sptr>();
 
   return data;
 }
 
-bool QWorkspaceDockView::askUserYesNo(const std::string &caption,
-                                      const std::string &message) const {
-  return QMessageBox::question(m_appParent, QString::fromStdString(caption),
-                               QString::fromStdString(message),
-                               QMessageBox::Yes,
-                               QMessageBox::No) == QMessageBox::Yes;
+bool WorkspaceTreeWidget::askUserYesNo(const std::string &caption,
+                                    const std::string &message) const {
+  return QMessageBox::question(
+             dynamic_cast<QWidget *>(this->parent()),
```

Comment:
You could use [`parentWidget`](http://doc.qt.io/archives/qt-4.8/qwidget.html#parentWidget) instead here to avoid the `dynamic_cast`.